rest_framework_simplejwt/serializers.py

```diff
@@ -112,11 +112,12 @@ def validate(self, attrs: dict[str, Any]) -> dict[str, str]:
         refresh = self.token_class(attrs["refresh"])
 
         user_id = refresh.payload.get(api_settings.USER_ID_CLAIM, None)
-        if user_id and (
-            user := get_user_model().objects.get(
-                **{api_settings.USER_ID_FIELD: user_id}
+        if user_id:
+            user = (
+                get_user_model()
+                .objects.filter(**{api_settings.USER_ID_FIELD: user_id})
+                .first()
```

Comment:
this may not be expected since the `USER_AUTHENTICATION_RULE` expects a user object, not None

https://github.com/jazzband/djangorestframework-simplejwt/blob/faeea49e33c971f28b505c2d8ba8384b56802f27/rest_framework_simplejwt/authentication.py#L170

Reply:
Since the `USER_AUTHENTICATION_RULE` method already [performs](https://github.com/jazzband/djangorestframework-simplejwt/blob/faeea49e33c971f28b505c2d8ba8384b56802f27/rest_framework_simplejwt/authentication.py#L178) the `user is not None` check, would adding another check outside the method introduce unnecessary code duplication?

Reply:
yea, the problem is the initial typing was typed assuming the user object is valid...

I mean that's my fault for not checking that. I guess this is fine, but we should also update the typing for the authentication rule?

Reply:
Makes sense, updated